src/tools/rbd/action/Nbd.cc

```diff
@@ -80,35 +80,60 @@ static int call_nbd_cmd(const po::variables_map &vm,
   return 0;
 }
 
+void get_device_specific_map_options(const std::string &help_suffix,
+                                     po::options_description *options) {
+  options->add_options()
+    ("device", po::value<std::string>(),
+     ("specify nbd device" + help_suffix).c_str())
+    ("nbds_max", po::value<std::string>(),
+     ("override module param nbds_max" + help_suffix).c_str())
+    ("max_part", po::value<std::string>(),
```

Comment:
Nit: when invoked from ```Kernel.cc```, it might be nice to change the optional to ```--nbd_(device|max_part|timeout)``` and only keep it ```--XYZ``` for the deprecated action. That way it's immediately clear which options are only valid for NBD maps.

Reply:
May be reuse `--options`, which we already use for krbd? I.e.:
```
rbd device map -t krbd -o rw,nocrc,mount_timeout=10 ...
rbd device map -t nbd -o nbd_device=256,timeout=10   ...
```

Reply:
Sorry, for `nbd` I meant something like:
```
rbd device map -t nbd -o device=/dev/nbd1,timeout=10
```

Reply:
@trociny That works for me -- just need to document the possible options somehow, though

Reply:
Or we could make vice versa: for krbd, instead of:
```
rbd device map -t krbd -o rw,nocrc,mount_timeout=10 ...
```
have:
```
rbd device map -t krbd --krbd_rw --krbd_nocrc --krbd_mount_timeout=10 ...
```
I just would like we have this consistent between different device types.

Reply:
Something about nearly a dozen ```--krbd_XYZ```-prefixed optionals makes me prefer the older style instead even though the new style is self-documenting. As a third possibility, perhaps all the driver-specific options can be added to the help only after you specify the type so that they are only accepted as valid optionals when applicable to the current device type:

```
rbd help device map --device-type nbd
... snip ... snip ... snip ...
  Optional arguments
    --device arg             specify nbd device +
    --nbds_max arg           override module param nbds_max +
    --max_part arg           override module param max_part +
    --timeout arg            set nbd request timeout (seconds) +
```

@idryomov thoughts?

Reply:
`rbd help device map --device-type nbd` looks too cumbersome to me.  I'd vote for reusing `-o/--options`.  It's not self-documenting, but we can make it so -- although personally I'd rather we improve `rbd(8)` man page.